### ISSUE TITLE
Python: Two classes in one file, second class documentation block copied into its class method block

### ISSUE THREAD

**Issue by spluttflob** [doxygen/doxygen#9158]
**Describe the bug**
A Python file contains two classes, each with some methods, all documented with docstrings.  Documentation is generated correctly for the first class.  The class documentation block for the second class in the file is copied into the block for the first method belonging to the second class. 

**Expected behavior**
The class documentation for each class should not be copied into the function documentation block for any method. 

**To Reproduce**
Please see attached .zip, in which the following triggers the apparent bug:
```.py
class Alpha:
    """! The class documentation of the first class. """
    def __init__(self):
        """! The first class's initializer. """
    def moo(self):
        """! Perhaps make a bovine sound? """
        
class Beta:
    """! The class documentation of the second class. """
    def __init__(self):
        """! The second class's constructor. """
    def meow(self):
        """! Make a feline sound, but not purring. """
```
**Screenshot**
The errant result:
![doxy-bug](https://user-images.githubusercontent.com/60046013/155654495-4c6cd0cc-09d0-41ab-924a-caba38830884.png)

**Version**
* Found on 1.8.17 running on Ubuntu 20.04.4 LTS, installed from Ubuntu repositories.
* Verified on freshly compiled 1.9.4.

**The Zip**
[doxy_bug.zip](https://github.com/doxygen/doxygen/files/8138494/doxy_bug.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9170

**Comment by spluttflob** [comment]
Tried the patch on my documentation, and it fixed the problem with no apparent side effects.  Many thanks!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
